perf/perf_cpu_hotplug.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2021 IBM
+# Author: Nageswara R Sastry <rnsastry@linux.ibm.com>
+
+import os
+import platform
+from avocado import Test
+from avocado.utils import cpu, distro, genio, process
+from avocado.utils.software_manager import SoftwareManager
+from avocado import skipIf
+
+IS_POWER_NV = 'PowerNV' in genio.read_file('/proc/cpuinfo').rstrip('\t\r\n\0')
+
+
+class perf_cpu_hotplug(Test):
+
+    """
+    This tests hv_24x7 and hv_gpci cpu_hotplug
+    :avocado: tags=perf,cpu_hotplug,events
+    """
+    # Initializing fail command list
+    fail_cmd = list()
+
+    def _check_required_files(self, event_type):
+        self.event_sysfs = "/sys/bus/event_source/devices/%s" % event_type
+        self.event_cpumask = "/sys/devices/%s/cpumask" % event_type
+        if os.path.exists("%s" % self.event_sysfs):
+            self.log.info('%s present' % event_type)
+            if event_type == "hv_24x7":
+                self.hv24x7_present = True
+                if os.path.exists("%s" % self.event_cpumask):
+                    self.log.info("%s cpumask present" % event_type)
+                    self.hv24x7_cpumask = True
+            elif event_type == "hv_gpci":
+                self.hvgpci_present = True
+                if os.path.exists("%s" % self.event_cpumask):
+                    self.log.info("%s cpumask present" % event_type)
+                    self.hvgpci_cpumask = True
+
+    @skipIf(IS_POWER_NV, "This test is supported on PowerVM environment")
+    def setUp(self):
+        """
+        Setup checks :
+        0. Processor should be ppc64.
+        1. Install perf package
+        2. Check for hv_24x7/hv_gpci cpumask
+        3. Offline the cpumask CPU and check cpumask moved to new CPU or not
+        """
+        smm = SoftwareManager()
+
+        detected_distro = distro.detect()
+        if 'ppc64' not in detected_distro.arch:
+            self.cancel("Processor is not PowerPC")
+
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in detected_distro.name:
+            deps.extend(['linux-tools-common', 'linux-tools-%s'
+                         % platform.uname()[2]])
+        elif detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(['perf', 'numactl'])
```

Comment:
any specific reason numactl is required for this test to run?

Reply:
Removed 'numactl'